tools/depends/configure.ac

```diff
@@ -418,7 +418,7 @@ case $host in
     platform_cxxflags+=" -arch $use_cpu -m$platform_min_version -stdlib=libc++"
     platform_includes="-isysroot $use_sdk_path"
     deps_dir="${sdk_name}_${use_cpu}-target-${build_type}"
-    AC_CHECK_LIB([z], [main], has_zlib=1, AC_MSG_WARN("No zlib support in toolchain. Will build libz."); has_zlib=0)
+    AC_SEARCH_LIBS([deflate],z, has_zlib=1, AC_MSG_WARN("No zlib support in toolchain. Will build libz."); has_zlib=0)
```

Comment:
This is wrong. As mariadb has to link zlib (because of openssl), it has to be fixed in the mariadb package.

Reply:
is there a reason not to have it as a linker flag for everything? We dont build zlib for apple platforms, as they have their own system version.

Reply:
@Rechi any further comment here?

Reply:
Because once the library which required it, doesn't need it anymore or gets remove this will be kept for no reason.

Reply:
It's used in like 7 other dependencies. But point taken. I'll change as requested.